### PR TITLE
feat: update momento client and improve error messages

### PR DESCRIPTION
**Pull request by brayniac**
Update the `momento` crate to the latest version and improve error messages returned in both client responses and those used for logging. This can help improve experience when trying to debug issues.

### CODE REVIEW

File: src/proxy/momento/src/klog.rs [hunk 1]
```diff
@@ -13,6 +15,9 @@ pub enum Status {
     Deleted = 7,
     NotFound = 8,
     NotStored = 9,
+    ClientError = 10,
```
Comment (thinkingfish) [inline src/proxy/momento/src/klog.rs:18]: Is there merit in assigning negative values to "abnormal" return statuses including errors and exceptions? May be easier to filter in post processing for different analyses.

File: src/proxy/momento/src/protocol/memcache/get.rs [hunk 2]
```diff
@@ -76,22 +76,22 @@ pub async fn get(
...
-            Ok(Err(MomentoError::LimitExceeded(_))) => {
-                BACKEND_EX.increment();
-                BACKEND_EX_RATE_LIMITED.increment();
-            }
             Ok(Err(e)) => {
                 // we got some error from the momento client
                 // log and incr stats and move on treating it
                 // as a miss
-                error!("error for get: {}", e);
+                error!("backend error for get: {}", e);
```
Comment (thinkingfish) [inline src/proxy/momento/src/protocol/memcache/get.rs:83]: Do we need to differentiate backend error from proxy error? These are used interchangeably right now but I think they don't always agree.

File: src/proxy/momento/src/klog.rs (hunk 1 above)
Comment (brayniac) [reply src/proxy/momento/src/klog.rs:18]: If we're willing to revisit the codes with the intent of making the klog easier to parse, maybe we want to use 3 digit codes like HTTP status codes so we can group things by cause as well as by specific status? Feels like we could bikeshed on this but I like the idea of updating the format to better accommodate use in backend, proxy, and even client scenarios.

File: src/proxy/momento/src/protocol/memcache/get.rs (hunk 2 above)
Comment (brayniac) [reply src/proxy/momento/src/protocol/memcache/get.rs:83]: I think a user will want to know what component is having an issue. We're limited in what we can provide in the memcached protocol - we can only provide messages for `CLIENT_ERROR` and `SERVER_ERROR`. The plain `ERROR` response does not allow a message. I think here, the best we can do is try to ascribe the fault to a component in the log message and return a `SERVER_ERROR` with some description for both proxy and backend errors.

For RESP, I think we should try to put a descriptive message in the `-ERR` response.

**Review by thinkingfish — APPROVED** [review]

**Pull request merged**